includes/manifest.inc

```diff
@@ -11,94 +11,145 @@
 function islandora_iiif_manifests_view_manifest_json(AbstractObject $object) {
   module_load_include('inc', 'islandora_iiif_manifests', 'includes/utilities');
   module_load_include('inc', 'islandora', 'includes/datastream');
-
+  
+  $data = "";
+  
   if (isset($object[ISLANDORA_IIIF_MANIFESTS_DATASTREAM_ID]) && islandora_datastream_access(ISLANDORA_VIEW_OBJECTS, $object[ISLANDORA_IIIF_MANIFESTS_DATASTREAM_ID])) {
     header('Access-Control-Allow-Origin: *');
-
+    
     islandora_view_datastream($object[ISLANDORA_IIIF_MANIFESTS_DATASTREAM_ID]);
   }
   else {
-    islandora_iiif_manifests_create_json_response('403 Forbidden', 'Forbidden');
+    if (!isset($object[ISLANDORA_IIIF_MANIFESTS_DATASTREAM_ID]))
+      $data = "No manifest found";
+    
+    if (islandora_datastream_access(ISLANDORA_VIEW_OBJECTS, $object[ISLANDORA_IIIF_MANIFESTS_DATASTREAM_ID]))
+      $data = "Forbidden";
+    
+    islandora_iiif_manifests_create_json_response('403 Forbidden', $data);
   }
 }
 
 /**
  * Response in JSON for the manifest.
  */
 function islandora_iiif_manifests_create_manifest_json(AbstractObject $object) {
-
+  
   module_load_include('inc', 'islandora', 'includes/utilities');
-
+  
   $output = islandora_iiif_manifests_build_manifest($object);
-
+  
   islandora_iiif_manifests_create_json_response('200 OK', $output);
 }
 
+
 /**
  * Response in JSON for the manifest.
  */
 function islandora_iiif_manifests_create_manifest_datastream(AbstractObject $object) {
-
+  
   module_load_include('inc', 'islandora', 'includes/utilities');
-
+  
   $output = islandora_iiif_manifests_build_manifest($object);
-
+  
   islandora_iiif_manifests_add_manifest_datastream($object, $output);
 }
 
 /**
  * Response in JSON for the canvas.
  */
 function islandora_iiif_manifests_create_canvas_json(AbstractObject $object) {
-
+  
   module_load_include('inc', 'islandora', 'includes/utilities');
-
+  
   $output = islandora_iiif_manifests_build_canvas($object);
-
+  
   islandora_iiif_manifests_create_json_response('200 OK', $output);
 }
 
+/**
+ * Response in JSON for the sub collection
+ */
+function islandora_iiif_manifests_create_collection_json(AbstractObject $object, $collection) {
+  
+  module_load_include('inc', 'islandora', 'includes/utilities');
+  
+  if (!isset($collection)){
+
+    islandora_iiif_manifests_create_json_response('403 Forbidden',"Collection not set" );
```

Comment:
If the collection is not set, isn't it a 400 Bad Request instead of 403 Forbidden?

Reply:
Yes you are right